ui/lib/src/controls.ts

```diff
@@ -64,14 +64,7 @@ export const boolPrefXhrToggle = (prefKey: string, val: boolean, effect: () => v
   });
 
 export function stepwiseScroll(inner: (e: WheelEvent, scroll: boolean) => void): (e: WheelEvent) => void {
-  let scrollTotal = 0;
-  return (e: WheelEvent) => {
-    scrollTotal += e.deltaY * (e.deltaMode ? 40 : 1);
-    if (Math.abs(scrollTotal) >= 4) {
-      inner(e, true);
-      scrollTotal = 0;
-    } else inner(e, false);
-  };
```

Comment:
I'm a bit surprised that it turns out we don't need any of that. Why did we rely on `e.deltaMode` and why were we wrong to do so?

Also,
```js
inner(e, !e.ctrlKey); // if touchpad zooming, e.ctrlKey is true
```
That is not documented by MDN: https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/ctrlKey

It might be a bit unsafe to rely on such undocumented behaviours.

Reply:
Here the `.ctrlKey` is documented: https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent
The table lists safari and webview (both on ios) as exceptions, but from what I understand the pinch to zoom doesn't trigger wheel events on them in the first place.

For the current logic with `e.deltaMode`, it tries to distinguish between mouse and trackpad usage, which is harder than distinguishing between specifically mouse scrolling and trackpad zooming. E.g., the answers [here] (https://stackoverflow.com/questions/10744645/detect-touchpad-vs-mouse-in-javascript) offer solutions, but a number of commenters say they get issues.

Reply:
OK, thanks! Let's try it out.